lua/tests/docgen/function_spec.lua

```diff
@@ -0,0 +1,476 @@
+            description = { "", "This function no documentation", "" },
+            ["field_list"] = {}, fields = {},
+            format = "function",
+            name = "x.bye",
+            ["parameter_list"] = {}, parameters = {}
+          },
+          ["x.good_evening"] = {
+            class = {},
+            description = { "", "This function some documentation", "" },
+            ["field_list"] = {}, fields = {},
+            format = "function",
+            name = "x.good_evening",
+            ["parameter_list"] = {}, parameters = {}
+          }
+        }
+      }, nodes)
+    end)
+
+    it('should get the nodes of a complex exported function', function()
+      local nodes = get_dedented_nodes [[
+        local x = {}
+
+        --- This function has documentation
+        ---@param abc string: Docs for abc
+        ---@param def string: Other docs for def
+        ---@param bxy number: Final docs
+        function x.hello(abc, def, bxy)
+          return abc .. def .. tostring(bxy)
+        end
+
+        return x
+      ]]
+      eq({
+        ["function_list"] = { "x.hello" },
+        functions = {
+          ["x.hello"] = {
+            class = {},
+            description = { "", "This function has documentation", "" },
+            ["field_list"] = {}, fields = {},
+            format = "function",
+            name = "x.hello",
+            ["parameter_list"] = { "abc", "def", "bxy" },
+            parameters = {
+              ["abc"] = { description = { "Docs for abc" }, name = "abc", type = "string" },
+              ["def"] = { description = { "Other docs for def" }, name = "def", type = "string" },
+              ["bxy"] = { description = { "Final docs" }, name = "bxy", type = "number" },
+            }
+          }
+        }
+      }, nodes)
+    end)
+  end)
+
+  describe('help output', function()
+    local function check_function_output(input, output, block)
+      local nodes = require('docgen').get_nodes(input)
+      local result = docgen_help.format(nodes)
+      block = block == nil and true or block
+      output = block and help_block(output) or ''
+
+      eq(vim.trim(output), vim.trim(result))
+    end
+    it('should export documented function', function()
+      check_function_output([[
+        local x = {}
+
+        --- This function has documentation
+        function x.hello()
+          return 5
+        end
+
+        return x]], [[
+        x.hello()                                                          *x.hello()*
+            This function has documentation]])
+    end)
+
+    it('should export this style as well function', function()
+      check_function_output([[
+        local x = {}
+
+        --- This function has documentation
+        x.hello = function()
+          return 5
+        end
+
+        return x]], [[
+        x.hello()                                                          *x.hello()*
+            This function has documentation]])
+    end)
```

Comment:
This format isn't working.

Its not picked up by the transformer even tho you have this:
```lua
transformers.function_statement = transformers._function
transformers.variable_declaration = transformers._function
```
The only difference in grammar here is that `function_name` has a `table_dot`. Should this make a difference? Because playground is picking it up

![image](https://user-images.githubusercontent.com/15233006/110254200-73370000-7f8e-11eb-93c9-98a0ef5628cd.png)

Reply:
So that never works in the current setup? I can work on fixing that if so

Reply:
Its weird because 
```lua
--- Live grep means grep as you type.
builtin.live_grep = require('telescope.builtin.files').live_grep
```
is being picked up with current master and this PR.

Difference in tree is that my test has `function` as value and `live_grep` a `function_call`